src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -281,24 +281,30 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             }
         }
 
-        public int Capacity {
+        public int Capacity
+        {
             get { return m_ChunkChars.Length + m_ChunkOffset; }
-            set {
-                if (value < 0) {
+            set
+            {
+                if (value < 0)
+                {
                     throw new ArgumentOutOfRangeException("value", Environment.GetResourceString("ArgumentOutOfRange_NegativeCapacity"));
                 }
-                if (value > MaxCapacity) {
+                if (value > MaxCapacity)
+                {
                     throw new ArgumentOutOfRangeException("value", Environment.GetResourceString("ArgumentOutOfRange_Capacity"));
                 }
-                if (value < Length) {
+                if (value < Length)
+                {
                     throw new ArgumentOutOfRangeException("value", Environment.GetResourceString("ArgumentOutOfRange_SmallCapacity"));
                 }
                 Contract.EndContractBlock();
 
-                if (Capacity != value) {
+                if (Capacity != value)
+                {
                     int newLen = value - m_ChunkOffset;
                     char[] newArray = new char[newLen];
-                    Array.Copy(m_ChunkChars, newArray, m_ChunkLength);
+                    Buffer.BlockCopy(m_ChunkChars, 0, newArray, 0, m_ChunkLength * sizeof(char));
```

Comment:
I just realized, maybe this could result in an integer overflow if `m_ChunkLength > int.MaxValue / 2`? Should I use `checked` here, or are arrays not allowed to be that big?